### PR TITLE
Update release workflow

### PR DESCRIPTION
**Pull request by gustavste**
Update release workflow

### CODE REVIEW

File: .github/workflows/prepare-release.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Prepare release
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-type:
+        type: choice
+        description: |
+          Release type
+          Use prepatch, preminor, premajor for the first "pre", to increment pre value use prerelease.
```
Comment (gnirb) [inline .github/workflows/prepare-release.yaml:10]: Nice, didn't know about this being supported by gh actions!

File: .github/workflows/prepare-release.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    runs-on: ubuntu-latest
+    env:
+      RELEASE_TYPE: ${{ github.event.inputs.release-type }}
+    steps:
+      - uses: actions/checkout@v3
+      # Configure Git
+      - name: Git configuration
+        run: |
+          git config user.name Release-Workflow
+          git config user.email github-actions@github.com
```
Comment (gnirb) [inline .github/workflows/prepare-release.yaml]: Not a big deal, but seems like there was some more detailed way of setting the email: https://github.com/orgs/community/discussions/26560

File: .github/workflows/release-and-publish.yaml [hunk 3]
```diff
@@ -102,7 +81,7 @@ jobs:
         with:
           tag_name: ${{ env.NEW_VERSION }}
           body: ${{ steps.get-changelog.outputs.changelog }}
```
Comment (gnirb) [inline .github/workflows/release-and-publish.yaml:84]: Nifty, didn't know about this either!

**Review by gnirb — APPROVED** [review]
lgtm, lets give it a try

File: .github/workflows/prepare-release.yaml (hunk 1 above)
Comment (tommythe) [reply .github/workflows/prepare-release.yaml:10]: Nice. Perhaps we will not use all variants and we end up just using 2.

File: .github/workflows/release-and-publish.yaml [hunk 4]
```diff
@@ -1,93 +1,72 @@
...
+          [[ ! $NEW_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]] && echo "Version '$NEW_VERSION' does not match regex" && exit 1
+      - name: Verify beta version regex
+        if: ${{ contains(env.NEW_VERSION, 'beta') == true }}
+        run: |
+          [[ ! $NEW_VERSION =~ ^v[0-9]+\.[0-9]+\.[0-9]+-beta\.[0-9]+$ ]] && echo "Version '$NEW_VERSION' does not match regex" && exit 1
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: "16.x"
+          node-version: "18.x"
```
Comment (tommythe) [inline .github/workflows/release-and-publish.yaml:30]: Maybe add npm caching to speed up builds, iff it makes sense. 

https://github.com/actions/setup-node#caching-global-packages-data

**Review by tommythe — APPROVED** [review]
Please consider adding a very short instruction in the docs about how to produce a new release

**Pull request merged**